src/daft-parquet/src/stream_reader.rs

```diff
@@ -696,26 +665,39 @@ pub async fn local_parquet_stream(
         metadata,
         chunk_size,
         io_stats,
-        semaphore.clone(),
     )?;
 
-    let (output_senders, output_receivers) = row_ranges
-        .iter()
-        .map(|_| tokio::sync::mpsc::channel(1))
-        .unzip::<_, _, Vec<_>, Vec<_>>();
+    // We use a semaphore to limit the number of concurrent row group deserialization tasks.
+    // Set the maximum number of concurrent tasks to ceil(number of available threads / columns).
+    let num_parallel_tasks = (std::thread::available_parallelism()
+        .unwrap_or(NonZeroUsize::new(2).unwrap())
+        .checked_mul(2.try_into().unwrap())
+        .unwrap()
+        .get() as f64
+        / max(schema_ref.fields.len(), 1) as f64)
+        .ceil() as usize
+        * 2;
```

Comment:
i think we have 1 too many `x 2`. we already have the `checked_mul`, did you mean to keep this in?

Reply:
Nope, no need the extra.